command/cat.go

```diff
@@ -32,46 +32,76 @@ var catCommand = &cli.Command{
 	Usage:              "print remote object's contents to stdout",
 	CustomHelpTemplate: catHelpTemplate,
 	Before: func(c *cli.Context) error {
+		op := c.Command.Name
+		fullCommand := givenCommand(c)
+
 		if c.Args().Len() != 1 {
-			return fmt.Errorf("expected only one argument")
+			err := fmt.Errorf("expected only one argument")
+			printError(fullCommand, op, err)
```

Comment:
how about defining validate function and logging these validation errors in only one place just like [cp](https://github.com/peak/s5cmd/blob/cf9fb156b02fca5af419121a7cef85e665e927bd/command/cp.go#L631)? 

```go
func validate(c *cli.Context) error {
	if c.Args().Len() != 1 {
		return fmt.Errorf("expected only one argument")
	}

	src, err := url.New(c.Args().Get(0))
	if err != nil {
		return err
	}

	if !src.IsRemote() {
		return fmt.Errorf("source must be a remote object")
	}

	if src.IsBucket() || src.IsPrefix() {
		return fmt.Errorf("remote source must be an object")
	}

	if src.HasGlob() {
		return fmt.Errorf("remote source %q can not contain glob characters", src)
	}

	return nil
}

// caller
err := validate(c)
if err != nil {
	printError(givenCommand(c), c.Command.Name, err)
}
return err

```